rwgame/main.cpp

```diff
@@ -1,9 +1,28 @@
 #include "RWGame.hpp"
+#include "SDL.h"
+#include <iostream>
 
-int main(int argc, char *argv[])
+int main(int argc, char* argv[])
 {
+	try {
+		RWGame game(argc, argv);
 
-	RWGame game( argc, argv );
+		return game.run();
+	} catch (std::runtime_error& ex) {
+		// Catch runtime_error as these are fatal issues the user may want to
+		// know about like corrupted files or GL initialisation failure.
+		// Catching other types (out_of_range, bad_alloc) would just make
+		// debugging them more difficult.
 
-	return game.run();
+		const char* kErrorTitle = "Fatal Error";
+
+		std::cerr << kErrorTitle << "\n" << ex.what() << std::endl;
```

Comment:
nit: this should use logging code like I said, it's totally non-standard compared to all other text we print. It works for now though